### PR TITLE
Read runtime storage value using `ext_get_runtime_storage`

### PR DESCRIPTION
**Pull request by ascjones**
Query runtime storage from within an ink! contract.

Corresponding substrate PR adding `ext_get_runtime_storage` : https://github.com/paritytech/substrate/pull/4108

**TODO**

- [x] Test it works with on chain contract
- [x] Implement env2 testenv
- [x] Remove ink 1.0 impls (only implement in env2/lang2)?

/cc @gautamdhameja @pepyakin

### CODE REVIEW

File: core/src/env2/env_access/mutable.rs [hunk 1]
```diff
@@ -333,6 +333,7 @@ where
     where
         R: scale::Decode,
     {
+        self.println("mutable::get_runtime_storage");
```
Comment (Robbepop) [inline core/src/env2/env_access/mutable.rs]: Can you try printing the key?

File: core/src/env2/env_access/mutable.rs [hunk 2]
```diff
@@ -333,6 +333,7 @@ where
...
         T::get_runtime_storage(&mut self.buffer, key)
```
Comment (Robbepop) [inline core/src/env2/env_access/mutable.rs]: Can you try printing the returned underlying buffer?

File: core/src/env/srml/srml_only/impls.rs [hunk 3]
```diff
@@ -245,4 +245,13 @@ where
         <Self as EnvTypes>::AccountId::decode(&mut &read_scratch_buffer()[..])
             .map_err(|_| CreateError)
     }
+
+    fn runtime_get_storage<U: Decode>(key: &[u8]) -> Option<Result<U, scale::Error>> {
```
[the remark this answers is not in the document]
Comment (ascjones) [reply core/src/env/srml/srml_only/impls.rs]: Already done

File: core/src/env2/srml/ext.rs [hunk 4]
```diff
@@ -87,6 +87,10 @@ pub fn get_storage(key: &[u8]) -> RetCode {
     unsafe { sys::ext_get_storage(key.as_ptr() as u32) }.into()
 }
 
+pub fn runtime_get_storage(key: &[u8]) -> RetCode {
+    unsafe { sys::ext_runtime_get_storage(key.as_ptr() as u32, key.len() as u32) }.into()
+}
+
```
[the remark this answers is not in the document]
Comment (ascjones) [reply core/src/env2/srml/ext.rs]: Already done - this was an old version

File: core/src/env2/env_access/mutable.rs (hunk 1 above)
Comment (ascjones) [reply core/src/env2/env_access/mutable.rs]: Fixed now

File: core/src/env2/env_access/mutable.rs (hunk 2 above)
Comment (ascjones) [reply core/src/env2/env_access/mutable.rs]: Fixed now

**Pull request merged**